libraries/RC_Channel/RC_Channel.h

```diff
@@ -11,7 +11,9 @@
 #include <AP_Math/AP_Math.h>
 #include <AP_Common/Bitmask.h>
 
+#ifndef NUM_RC_CHANNELS
```

Comment:
We don't want to bare something like `NUM_RC_CHANNELS` to the outside world.

This would really need to be something like this:
```
#ifdef AP_RC_CHANNEL_COUNT
#define NUM_RC_CHANNELS AP_RC_CHANNEL_COUNT
#else 
#define NUM_RC_CHANNELS 16
#endif
```